src/main/java/spireMapOverhaul/zoneInterfaces/EncounterModifyingZone.java

```diff
@@ -0,0 +1,122 @@
+     * Note that as part of this method, you can check the current act (use ActUtil.getRealActNum()), the room type
+     * (e.g. AbstractDungeon.getCurrRoom instanceof MonsterRoomElite), etc.
+     * The x position of the monster is automatically determined and will be to the left of all the enemies already present,
+     * with space automatically allocated based on the size of the monster's hitbox. (Any x position set as part of the
+     * monster's constructor will be overwritten.)
+     * Keep the number of size of additional monsters small to avoid them taking up so much space that they overlap with
+     * the player when added to fights that already have many enemies.
+     * @return The monsters to add to the encounter.
+     */
+    default List<AbstractMonster> getAdditionalMonsters() { return null; }
+
+    /**
+     * Defines a new encounter in a zone.
+     * An encounter requires an ID, which should be the same as the monster ID for single enemy encounters. This is
+     * needed by BaseMod (and can be used to fight your new encounter from the console with the fight command).
+     * An encounter also requires an act number, and will only show up in that act.
+     * The enemies in an encounter are defined by a function that returns either a single monster or a MonsterGroup.
+     * An encounter may have a special name. If no name is provided, the encounter will be named based on the enemies.
+     */
+    public static class ZoneEncounter {
+        private String id;
+        private int actNum;
+        private Supplier<MonsterGroup> f;
+        private String name;
+
+        public ZoneEncounter(String id, int actNum, BaseMod.GetMonster f) {
+            this(id, actNum, () -> new MonsterGroup(f.get()), null);
+        }
+
+        public ZoneEncounter(String id, int actNum, BaseMod.GetMonsterGroup f) {
+            this(id, actNum, f, null);
+        }
+
+        public ZoneEncounter(String id, int actNum, BaseMod.GetMonster f, String name) {
+            this(id, actNum, () -> new MonsterGroup(f.get()), name);
+        }
+
+        public ZoneEncounter(String id, int actNum, BaseMod.GetMonsterGroup f, String name) {
+            this.id = id;
+            this.actNum = actNum;
+            this.f = f::get;
+            this.name = name;
+        }
+
+        public String getID() {
+            return id;
+        }
+
+        public int getActNum() {
+            return actNum;
+        }
+
+        public Supplier<MonsterGroup> getMonsterSupplier() {
+            return f;
+        }
+
+        public String getName() {
+            return name;
+        }
+    }
+
+    /**
+     * Lets you change the encounter however you want.
+     * You should not need to use this unless you're doing something unusual.
+     * @param monsterGroup The encounter.
+     * @return The changed encounter.
+     */
+    default MonsterGroup changeEncounter(MonsterGroup monsterGroup) { return monsterGroup; }
```

Comment:
Do you have an example for when you'd use this?

Reply:
All the examples I thought of are somewhat contrived:
* You want to do your own enemy positioning
* You want to replace a specific enemy with a different enemy
* You want to implement something that does your own tracking of encounters to decide what to replace each encounter with

This was the last thing I added, and I saw it as an fallback in case some zone needed to do something that wasn't anticipated by the rest of the API.